webrender/src/resource_cache.rs

```diff
@@ -935,6 +936,20 @@ impl ResourceCache {
                 let (actual_width, actual_height) =
                     compute_tile_size(image_descriptor, tile_size, tile);
 
+                if let Some(dirty) = dirty_rect {
+                    dirty_rect = dirty
```

Comment:
`dirty_rect` being `None` means that the whole image needs to be updated. In this case though, if there is zero intersection with the tile, we don't want any updates at all. I.e.:
```rust
if let Some(ref mut dirty) = dirty_rect {
  *dirty = match dirty.intersection(...) {
    Some(r) => {...; r}
    None => continue,
  };
}
```